cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -109,14 +109,15 @@ def upload_data(
             data["frame_filter"] = f"step={params.get('frame_step')}"
 
         if resource_type in [ResourceType.REMOTE, ResourceType.SHARE]:
+            str_resources = []
+
             for resource in resources:
-                if not isinstance(resource, str):
-                    raise TypeError(f"resources: expected instances of str, got {type(resource)}")
+                str_resources.append(os.fspath(resource))
```

Comment:
This could now be replaced with `str_resources = list(map(os.fspath, resources))`.